etc/kayobe/ansible/pulp_upload_image.yml

```diff
@@ -0,0 +1,135 @@
+    repository_name: "kayobe-images-{{ openstack_release }}-{{ os_distribution }}-{{ os_release }}"
+  tasks:
+    - name: Debug - create fake image directory # TODO: remove
+      ansible.builtin.file:
+        path: "{{ image_path }}"
+        state: directory
+        mode: '0766'
+      become: true
+
+    - name: Debug - create fake image file # TODO: remove
+      copy:
+        content: This is the image file content version 6
+        dest: "{{ image_path }}/fake-image.qcow2"
+        mode: '0766'
+      become: true
+
+    - name: Get filename
+      find:
+        paths: "{{ image_path }}"
+        patterns: '*.qcow2'
+      register: found_files
+      become: true # TODO: remove
+
+    - name: Upload an artifact
+      pulp.squeezer.artifact:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        file: "{{ found_files.files[0].path }}"
+        state: present
+      become: true # TODO: remove
+
+    - name: Get sha256 hash
+      ansible.builtin.stat:
+        path: "{{ found_files.files[0].path }}"
+        checksum_algorithm: sha256
+      register: file_stats
+      become: true # TODO: remove
+
+    - name: Create file content from artifact
+      pulp.squeezer.file_content:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        sha256: "{{ file_stats.stat.checksum }}"
+        relative_path: "{{ found_files.files[0].path | basename }}"
+        state: present
+
+    - name: Ensure file repo exists
+      pulp.squeezer.file_repository:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        name: "{{ repository_name }}"
+        state: present
+
+    - name: Add content to file repo
+      pulp.squeezer.file_repository_content:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        repository: "{{ repository_name }}"
+        present_content:
+          - relative_path: "{{ found_files.files[0].path | basename }}"
+            sha256: "{{ file_stats.stat.checksum }}"
+      register: repo_details
+
+    # Squeezer wants the version as an int. The previous play only returns the
+    # href of the latest version.
+    # The version can be returned by querying that href, however the last
+    # number in the href is the version. regex_search is used to extract it
+    - name: Create a new publication to point to this version
+      pulp.squeezer.file_publication:
+        pulp_url: "{{ remote_pulp_url }}"
+        username: "{{ remote_pulp_username }}"
+        password: "{{ remote_pulp_password }}"
+        repository: "{{ repository_name }}"
+        state: present
+        version: "{{ repo_details.repository_version | regex_search('(\\d+)(?!.*\\d)') }}"
```

Comment:
Won't this just be the latest version? So we shouldn't need the following task

Reply:
Good point, I've removed the other one